artemis-commons/src/main/java/org/apache/activemq/artemis/utils/actors/ProcessorBase.java

```diff
@@ -109,12 +125,49 @@ public final boolean isFlushed() {
       return stateUpdater.get(this) == STATE_NOT_RUNNING;
    }
 
-   protected void task(T command) {
-      tasks.add(command);
```

Comment:
-1.... lets just fix the offending part?

Reply:
It is not only the flush that has issues but the execute too: take a look at the tests I've added :)

Reply:
you've added tests that are simulating bugs at the code.. I would rather fix bugs than protect the executor from them.